File: Engr-T.stark/insertInterval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval)     {
        vector<vector<int>> ans;
        
        for(int i=0; i<intervals.size(); i++)
        {
            if(newInterval[1] < intervals[i][0])
            {
                ans.push_back(newInterval);
                newInterval = intervals[i];
            }
            else if(newInterval[0] > intervals[i][1])
            {
                ans.push_back(intervals[i]);
            }
            else if(newInterval[0] <= intervals[i][1] || newInterval[1] >= intervals[i][0])
            {
                newInterval[0] = min(intervals[i][0], newInterval[0]);
                newInterval[1] = max(intervals[i][1], newInterval[1]);
            }
        }
        
        ans.push_back(newInterval);
        
        return ans;
    }
};

File: Richie/Insert interval/insert-interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        output = []
        for i in range(len(intervals)):
            if newInterval[1] < intervals[i][0]:
                output.append(newInterval)
                return output + intervals[i:]
            elif newInterval[0] > intervals[i][1]:
                output.append(intervals[i])
            else:
                newInterval =[min(newInterval[0], intervals[i][0]),max(newInterval[1], intervals[i][1])]
                
        output.append(newInterval)
        return output
